### PR TITLE
Improve health check config test2

### PR DESCRIPTION
**Pull request by Devamparikh**
## Description
Fix https://github.com/eclipse/jkube/issues/1374: Replaced the expected annotation with Assert.assertThrows in HealthCheckConfigTest.java for exceptions verification with improvments.

## Type of change

 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [ ] Feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change
 - [x] Chore (non-breaking change which doesn't affect codebase;
   test, version modification, documentation, etc.)

## Checklist
 - [x] I have read the [contributing guidelines](https://www.eclipse.org/jkube/contributing)
 - [x] I signed-off my commit with a user that has signed the [Eclipse Contributor Agreement](https://www.eclipse.org/legal/ECA.php)
 - [x] I Added [CHANGELOG](../CHANGELOG.md) entry
 - [x] I have implemented unit tests to cover my changes
 - [x] I have updated the [documentation](../kubernetes-maven-plugin/doc) accordingly
 - [x] No new bugs, code smells, etc. in [SonarCloud](https://sonarcloud.io/dashboard?id=jkubeio_jkube) report
 - [x] I tested my code in Kubernetes
 - [x] I tested my code in OpenShift

Do let me know if something goes wrong.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -37,7 +37,6 @@ Usage:
 * Fix #1308: Add documentation for DeploymentConfigEnricher
 * Fix #1310: Add documentation + gradle integration test for ImageChangeTriggerEnricher
 * Fix #1325: `jkube.enricher.jkube-name.name` doesn't modify `.metadata.name` for generated manifests
-* Fix #1362: VolumePermissionEnricher : Replace iteration with bulk Collection.addAll call
```
Comment (rohanKanojia) [inline CHANGELOG.md:40]: Why is this change required?

**Review by rohanKanojia — APPROVED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (Devamparikh) [reply CHANGELOG.md:40]: Do i need to write there `Fix https://github.com/eclipse/jkube/issues/1374: Replaced the expected annotation with Assert.assertThrows in HealthCheckConfigTest.java for exceptions verification`?? Or it is not required?

Comment (rohanKanojia) [reply CHANGELOG.md:40]: Ideally, we track changes via changelog. If you feel like doing it, you can add.
